portability-core/src/main/java/org/dataportabilityproject/serviceProviders/google/contacts/VCardToGoogleContactConverter.java

```diff
@@ -69,20 +79,46 @@ private static Name convertToGoogleName(StructuredName vCardName) {
     Name name = new Name();
     name.setFamilyName(vCardName.getFamily());
     name.setGivenName(vCardName.getGiven());
+
+    FieldMetadata fieldMetadata = new FieldMetadata();
     if (vCardName.getAltId() == null) {
-      name.setMetadata(PRIMARY_FIELD_METADATA);
+      fieldMetadata.setPrimary(true);
     } else {
-      name.setMetadata(SECONDARY_FIELD_METADATA);
+      fieldMetadata.setPrimary(false);
     }
+
+    String vCardNameSource = vCardName.getParameter(SOURCE_PARAM_NAME_TYPE);
+    if (vCardNameSource == null || vCardNameSource.equals(CONTACT_SOURCE_TYPE)) {
```

Comment:
This seems to contradict the comment above about adding null source. Looks like you are adding null sources here but the comment indicates you're not sure if you can?

Reply:
Whoops, good catch!